### PR TITLE
Add support for CSI BAM index files (REVIEW)

### PR DESCRIPTION
**Pull request by nathanhaigh**
Most things are now in place. Although I still need to implement support for the bgzip compressed CSI index. If you want to test with the current code you'll need to:

```bash
samtools index -c my.bam
gunzip -dc my.bam.csi > my.bam.csi.uncompressed
mv my.bam.csi.uncompressed my.bam.csi
```

I'll try to make a further commit to this pull request next week which will add support for the compressed natively compressed `CSI` index.

### CODE REVIEW

File: js/bam/bam.js [hunk 1]
```diff
@@ -75,36 +78,84 @@ function makeBam(data, bai, callback) {
...
-                var nchnk = readInt(uncba, p+4);
-                p += 8 + (nchnk * 16);
-            }
-            var nintv = readInt(uncba, p); p += 4;
-            p += (nintv * 8);
-            if (nbin > 0) {
-                bam.indices[ref] = new Uint8Array(header, blockStart, p - blockStart);
-            }                     
-        }
+	var uncba = new Uint8Array(header);
```
Comment (nathanhaigh) [inline js/bam/bam.js]: In the case we have a `CSI` file, we need to unbgzip the index before we can successfully continue. I'm still trying to figure out the best place to do this.

**Comment by nathanhaigh** [comment]
@MariaNattestad This should now be ready for review/testing. Please try a CSI indexed BAM file:

```bash
samtools index -c my.bam
```

**Comment by MariaNattestad** [comment]
Thanks Nathan! I am at a conference this week, so I will review this when I get back.

File: js/bam/bam.js [hunk 2]
```diff
@@ -75,36 +78,89 @@ function makeBam(data, bai, callback) {
...
-            var nintv = readInt(uncba, p); p += 4;
-            p += (nintv * 8);
-            if (nbin > 0) {
-                bam.indices[ref] = new Uint8Array(header, blockStart, p - blockStart);
-            }                     
-        }
+	var unc = header;
+	if (/[^.]+$/.exec(bam.index.blob.name)[0] == 'csi') {
+		// csi extensions need decompression
+		unc = unbgzf(header);
```
Comment (nathanhaigh) [inline js/bam/bam.js]: This seems to work with small `CSI` files but not larger ones. So this needs looking into further.

**Pull request merged**